pelican/tests/test_generators.py

```diff
@@ -405,6 +405,103 @@ def test_per_article_template(self):
         self.assertIn(custom_template, self.articles)
         self.assertIn(standard_template, self.articles)
 
+    def test_period_archives_context(self):
+        """Test correctness of the period_archives context values."""
+
+        old_locale = locale.setlocale(locale.LC_ALL)
+        locale.setlocale(locale.LC_ALL, 'C')
+        settings = get_settings()
+        settings['CACHE_PATH'] = self.temp_cache
+
+        # No period archives enabled:
+        context = get_context(settings)
+        generator = ArticlesGenerator(
+            context=context, settings=settings,
+            path=CONTENT_DIR, theme=settings['THEME'], output_path=None)
+        generator.generate_context()
+        period_archives = generator.context['period_archives']
+        self.assertEqual(len(period_archives.items()), 0)
+
+        # Year archives enabled:
+        settings['YEAR_ARCHIVE_SAVE_AS'] = 'posts/{date:%Y}/index.html'
+        settings['YEAR_ARCHIVE_URL'] = 'posts/{date:%Y}/'
+        context = get_context(settings)
+        generator = ArticlesGenerator(
+            context=context, settings=settings,
+            path=CONTENT_DIR, theme=settings['THEME'], output_path=None)
+        generator.generate_context()
+        period_archives = generator.context['period_archives']
+        self.assertEqual(len(period_archives.items()), 1)
+        self.assertIn('year', period_archives.keys())
```

Comment:
nit: I'd just self.assertEqual, i.e.

```
self.assertEqual(
  {'year': {'period': [1970, 2014]}},
  period_archives)
```

or whatever the right comparison is.

It tests more, and makes it clearer what period_archives looks like.

Reply:
I can understand this test style. `period_archives` contains actual `Article` objects. Not really possible to test complete equality.

Reply:
But I agree, so I changed them as close to that as possible :).